### PR TITLE
[TS] LPP-47463 > LPS-171451 Cannot upload a file in a Select Document modal window inside a Form.

### PR DESCRIPTION
**Pull request by ces-quanhuynh**
**Motivation**
This work solves the problem reported in the [LPS-171451](https://issues.liferay.com/browse/LPS-171451) ticket.
cc: @TaiPhamLR 

**1. Cause:** The upload files are stored in the default folder for Document and Media upload files. So they are not listed in the Form-field-upload-folder. The folder was null so it return to default folderId (https://github.com/liferay/liferay-portal-ee/blob/470587ce0ffccb77ff770cb15bb88e486cd0f447/modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/toolbar/contributor/helper/MenuItemProvider.java#L98) 
**2. What we did to solve this problem:**
We should create a attribute to stored folderId for Form-field-upload-folder and set it to the request. So when it get the folderId of the request, they can get the right folder for Form-field-upload-folder (https://github.com/liferay/liferay-portal-ee/blob/470587ce0ffccb77ff770cb15bb88e486cd0f447/modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/portlet/toolbar/contributor/helper/DLPortletToolbarContributorHelper.java#L46)

**Steps to reproduce**
Go to default site > Contents & Data > Forms > + (New Form)
Drag Upload fields from the sidebar and publish
Go to Site Builder > Pages > + > Page > Blank
Create New Page and add Form Widget
Click (⋮)Options on right side > Configuration
Select Form name in Step1 and Save
Close Form-Configuration window and Publish
Go to page in step3
Click Select button upload a new file
Click +(New) > File Upload
(Don't Use "Drag & Drop Your Files or Browse to Upload" function)
Select File and Publish
Note the uploaded file is not listed
Actual result
No file is uploaded. 

Expected result 
File is uploaded successfully and can be selected for the form.

### CODE REVIEW

**Comment by TaiPhamLR** [comment]
Hi @quanhuynhces, I found the method [DDMUserPersonalFolderItemSelectorViewDisplayContext#getUploadURL](https://github.com/TaiPhamLR/liferay-portal/blob/71f5759b7bd1276c2a0eee4aa4a53eb67292ca73/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMUserPersonalFolderItemSelectorViewDisplayContext.java#L150) also contains `folderId`. So you can get `folderId` from  at ItemSelectorRepositoryEntryManagementToolbarDisplayContext without adding folderId property to `liferay-item-selector` taglib. How do you think about my opinion? please let me know. 
Thanks.

File: modules/apps/item-selector/item-selector-taglib/src/main/java/com/liferay/item/selector/taglib/internal/display/context/ItemSelectorRepositoryEntryManagementToolbarDisplayContext.java [hunk 1]
```diff
@@ -95,6 +100,22 @@ public CreationMenu getCreationMenu() {
...
+		try {
+			folder = DLAppLocalServiceUtil.getFolder(folderId);
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException);
+		}
+
+		_liferayPortletRequest.setAttribute(
+			WebKeys.DOCUMENT_LIBRARY_FOLDER, folder);
+
```
Comment (TaiPhamLR) [inline modules/apps/item-selector/item-selector-taglib/src/main/java/com/liferay/item/selector/taglib/internal/display/context/ItemSelectorRepositoryEntryManagementToolbarDisplayContext.java:105]: Hi @quanhuynhces , you can add a private method to get `folderId` like this:
```
private String _getFolderId() {
		 PortletURL uploadURL = (PortletURL) 
			_httpServletRequest.getAttribute(
				"liferay-item-selector:repository-entry-browser:uploadURL");
		 return ((LiferayPortletURL)uploadURL).getParameter("folderId");
}
```

Comment (quanhuynhces) [reply modules/apps/item-selector/item-selector-taglib/src/main/java/com/liferay/item/selector/taglib/internal/display/context/ItemSelectorRepositoryEntryManagementToolbarDisplayContext.java:105]: @TaiPhamLR, thank you. I've updated the code follow your comment and I think we should create _getFolder() method for cleaner.

Comment (ces-quanhuynh) [reply modules/apps/item-selector/item-selector-taglib/src/main/java/com/liferay/item/selector/taglib/internal/display/context/ItemSelectorRepositoryEntryManagementToolbarDisplayContext.java:105]: Thank you @TaiPhamLR. Following your comment but I change a little bit to getFolder() method. Is this more suitable, please let me know.

**Comment by ces-quanhuynh** [comment]
> Hi @quanhuynhces, I found the method [DDMUserPersonalFolderItemSelectorViewDisplayContext#getUploadURL](https://github.com/TaiPhamLR/liferay-portal/blob/71f5759b7bd1276c2a0eee4aa4a53eb67292ca73/modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/display/context/DDMUserPersonalFolderItemSelectorViewDisplayContext.java#L150) also contains `folderId`. So you can get `folderId` from at ItemSelectorRepositoryEntryManagementToolbarDisplayContext without adding folderId property to `liferay-item-selector` taglib. How do you think about my opinion? please let me know.
> Thanks.

@TaiPhamLR I think that the when we create the folderId property it is easier and cleaner. Because when we get folderId from UploadURL, we have to handle the URL to get folderId and set to the request(https://github.com/quanhuynhces/liferay-portal/blob/8618a3437b6fca76e6f909e8aad4fe0b17da0649/modules/apps/item-selector/item-selector-taglib/src/main/java/com/liferay/item/selector/taglib/servlet/taglib/RepositoryEntryBrowserTag.java#L361).

**Comment by TaiPhamLR** [comment]
Ok, that's good.
forward to: https://github.com/liferay-forms/liferay-portal/pull/2078

**Pull request closed without merging**